gm4_sweethearts/data/gm4_sweethearts/functions/kill_donor.mcfunction

```diff
@@ -0,0 +1,10 @@
+# @s = sneaking player donating health
+# run from transfer_donor
+
+execute store result score $deathmessage gm4_sh_data run gamerule showDeathMessages
+gamerule showDeathMessages false
+advancement grant @s only gm4:sweethearts
+tellraw @a [{"text":"","color":"white"},{"selector":"@s"},{"text":" was shot through the heart and "},{"selector":"@p[tag=gm4_sh_recipient]"},{"text":" was to blame"}]
```

Comment:
Death message is a bit length. Why not: "@s gave their last heart to @p[tag=gm4_sh_recipient]" to mimic the advancement description?

Reply:
The message was thought up by JP as a reference to the song, I quite like it but we can change it if people prefer a different one

Reply:
I didn't know the song reference :/